lib/core/importer.ts

```diff
@@ -1,40 +1,70 @@
 import { glob } from 'glob';
 import { resolve } from 'path';
-import { AUTO_INJECTABLE_WATERMARK } from '../interfaces';
+import {
+  AUTO_CONTROLLER_WATERMARK,
+  AUTO_INJECTABLE_WATERMARK,
+} from '../interfaces';
 
 type ClassType = new (...args: any[]) => any;
 
-export interface AutoImportResult {
+export interface AutoProviders {
   providers: ClassType[];
 }
 
+export interface AutoControllers {
+  controllers: ClassType[];
+}
+
+// TODO: refactoring duplicate code
 export class Importer {
   constructor(private readonly patterns: string[]) {}
 
-  static async load(patterns: string[]): Promise<AutoImportResult> {
+  static async loadProviders(patterns: string[]): Promise<AutoProviders> {
     const importer = new Importer(patterns);
     const pathNames = await importer.matchGlob();
     const results = await Promise.all(
-      pathNames.map((pathName) => importer.importFile(pathName)),
+      pathNames.map((pathName) => importer.importProvider(pathName)),
     );
     return {
       providers: results.flatMap((result) => result.providers),
     };
   }
 
-  public async importFile(pathName: string): Promise<AutoImportResult> {
+  static async loadControllers(patterns: string[]): Promise<AutoControllers> {
+    const importer = new Importer(patterns);
+    const pathNames = await importer.matchGlob();
+    const results = await Promise.all(
+      pathNames.map((pathName) => importer.importController(pathName)),
+    );
+    return {
+      controllers: results.flatMap((result) => result.controllers),
+    };
+  }
+
+  public async importProvider(pathName: string): Promise<AutoProviders> {
```

Comment:
Rather than creating a new function called load controllers, why not just add the Controller in the load function?

Then loadProviders would be a load function again.

And AutoImportResult would look like this
```typescript
export interface AutoImportResult {
  providers: ClassType[];
  controllers: ClassType[];
}
```